### PR TITLE
Developer Workflow Documentation testing

### PR DESCRIPTION
**Pull request by mts299**
# Scope

Documentation focuses on how to test PRs in GitHub.

Please review if this makes sense.

### CODE REVIEW

File: docs/dev/testing.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+`pip install git+https://github.com/superdarn/rst@<branchname>`
+Make sure to install any new instances of RST, or branches that are not master in a virtual environment to avoid mixing RST versions! 
+
+
+If you wish to clone RST for multi-branch comparison or testing in the future: 
+
+1. Clone the [RST repository](https://github.com/SuperDARN/rst.git) by typing 
+`git clone https://github.com/SuperDARN/rst.git` into the terminal or command line.
+
+2. Then checkout the branch you need to test or test against. The author of the Pull Request should give you some information on how to test the code and if you need to checkout other branches to compare results. To checkout a branch you want to test use this git commands:\
```
Comment (ecbland) [inline docs/dev/testing.md:29]:
```suggestion
2. Then "checkout" the branch you need to test. The author of the Pull Request should give you some information on how to test the code, and if you need to checkout other branches to compare results. To checkout the branch you want to test:
```

File: docs/dev/testing.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+1. Clone the [RST repository](https://github.com/SuperDARN/rst.git) by typing 
+`git clone https://github.com/SuperDARN/rst.git` into the terminal or command line.
+
+2. Then checkout the branch you need to test or test against. The author of the Pull Request should give you some information on how to test the code and if you need to checkout other branches to compare results. To checkout a branch you want to test use this git commands:\ 
+
+        git fetch 
+        git checkout <branch name>
+        git pull origin <branch name>
+    These lines *fetch* metadata on any new branches made, *checkout* the right branch you want to test and then *pull* changes to that branch so you're not behind any new changes made.\
```
Comment (ecbland) [inline docs/dev/testing.md:34]:
```suggestion
    These lines *fetch* metadata on any new branches made, *checkout* the branch you want to test, and then *pull* changes to that branch to make sure you have the latest version.
```

File: docs/dev/testing.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  
+    !!! Note 
+        if you are unaware what the branch name is at the top of the page below the Pull Request title and number shows the branch name and which branch it will be merged into.
```
Comment (ecbland) [inline docs/dev/testing.md:37]: Is this a bit clearer?
```suggestion
    !!! Note 
        If you cannot determine the name of the branch you should be testing, look at the top of the page below the Pull Request title. Here you will see the branch name and also the "base branch" that the changes  will be merged to.
```

File: docs/dev/testing.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+        git fetch 
+        git checkout <branch name>
+        git pull origin <branch name>
+    These lines *fetch* metadata on any new branches made, *checkout* the right branch you want to test and then *pull* changes to that branch so you're not behind any new changes made.\
+  
+    !!! Note 
+        if you are unaware what the branch name is at the top of the page below the Pull Request title and number shows the branch name and which branch it will be merged into. 
+
+3. From here you just need to build and compile the code. Run the normal [installation](../user/install.md)
```
Comment (ecbland) [inline docs/dev/testing.md:39]:
```suggestion
3. Build and compile the code using the normal RST [installation](../user/install.md) procedure.
```

File: docs/dev/testing.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+      
+        make.build && make.code
+
+5. Once this is complete follow any test cases the documentation and Pull Request the author describes.
```
Comment (ecbland) [inline docs/dev/testing.md:43]:
```suggestion
5. Follow any testing instructions provided by the Pull Request author, and any examples in the documentation. Where relevant, make up your own examples to check that the code works in other common situations (e.g. both new- and old-format files)
```

File: docs/dev/testing.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  
+    !!! Note 
+        if you are unaware what the branch name is at the top of the page below the Pull Request title and number shows the branch name and which branch it will be merged into. 
+
+3. From here you just need to build and compile the code. Run the normal [installation](../user/install.md)
+      
+        make.build && make.code
+
+5. Once this is complete follow any test cases the documentation and Pull Request the author describes. 
+Report anything in the Pull Request comments about how you tested the code, what your output is, and any feedback, or information.
```
Comment (ecbland) [inline docs/dev/testing.md:44]:
```suggestion
Report anything in the Pull Request comments about how you tested the code, what your output is, and any feedback or other information.
```

File: docs/dev/testing.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+Note: To suggest changes to and/or comment on certain sections of code, navigate to the Files Changed tab where it is possible to select multiple lines of code by clicking and dragging the '+' symbol next to line of code.
+6. Once you are satisfied with the Pull Request, **approve** it by clicking on the top `Files changed` then clicking Review changes and select *Approve*. Make sure to comment what you did in testing for your approval. 
+
+!!! Note
+    If the Pull Request has more than two approvals (one for code review and another for testing) with no comments to respond to, then please merge the code. Merge button is at the bottom of the Pull Request page.
```
Comment (ecbland) [inline docs/dev/testing.md:49]: Adding another note about a common situation!
```suggestion
!!! Note
    If the Pull Request has more than two approvals (one for code review and another for testing) with no comments to respond to, then please merge the code. Merge button is at the bottom of the Pull Request page. 
    
!!! Note
    If you identify a new bug during testing that is not related to the current Pull Request, report this bug by opening a new Issue.
```

**Review by ecbland — COMMENTED** [review]
A few minor suggestions, but this looks good!

**Review by mts299 — COMMENTED** [review]
Need  to look over this again apparently, python wording still exists ... whoops

**Comment by mts299** [comment]
@ecbland updated the documentation

**Review by ecbland — APPROVED** [review]

**Pull request merged**